### DISCUSSION TITLE
20240425 Weekly Development Meeting

### DISCUSSION THREAD

**Discussion by josecelano** [nautilus-cyberneering/.github#91] [Meetings]
- Time:  9:30 in Canary Islands (WET (UTC+0)).
- Time: 10:30 in Vaduz (CET (UTC+1)).

Location:
<https://meeting.blockfinance-eco.li/NautilusCyberneeringWeeklyDevMeeting>

Expected duration: 2 to 4 hours with breaks

Communicated attendants:

- Cameron Garnham(@da2ce7)  - 
- Jose Celano(@josecelano) - x
- Constantin Bosse(@cgbosse) - x
- Mario Melchor(@mario-nt) - 
- Graeme Byrne(@grmbyrn) - x
- Mick van Dijke(@mickvandijke) - 
- Iván Santana(@ivan9989-pkj) - x
- Ismael Torrado(@Lepisma07) - x
- Borja (@BorjaPractica) - x

-------------------------------------------------------------

## Agenda:

1. Everyone will take their turn and:
- Go over his notes added as comments for the meeting which will have been added in advance to give an update of all the work done during the previous week indicating.
	- The issues worked on
	- The progress made on them
	- Other things done
- Inform of any faced challenges and request help if necessary to work on specific issues.
- Highlight the encountered difficulties, especially if there is a risk of them causing delays or affecting end product's quality.
- Communicate what will be the next week's tasks.
- List any new topics for further discussion in an additional Technical meeting that he noted for future discussion in comments to this discussion either before the meeting or after his turn so that they get considered.

During the meeting every team member will take their turns and follow this sequence.
The topics for further discussion will be noted in advance by each the team member and to the meetings agenda.

2. At the end of the meeting if there any new technical topics to be discussed further it will be decided to
set another meeting date for the discussion of these or to hold the meeting hereafter.

Also the persons to be part of the meeting will be decided on so that the others will be free to continue their own work.

**Comment by josecelano** [comment]
### Torrust Tracker

My main focus was the configuration. [I was working on adding early validation to the configuration](https://github.com/torrust/torrust-tracker/pull/790), but I realized most of that validation could be done using reach types instead of primitive types. So I decided to [start the migration to Figment](https://github.com/torrust/torrust-tracker/pull/808) first, which we wanted to do anyway. However, I have not had too much time for this. This week, there were more meetings and PR reviews. And I have also been busy [trying to fix problems with the live demo server](https://github.com/torrust/torrust-demo).

New Issues: 

- [ ] https://github.com/torrust/torrust-tracker/issues/811

Issues:

- [ ] https://github.com/torrust/torrust-tracker/issues/612
- [ ] https://github.com/torrust/torrust-tracker/issues/521

PRs:

- **WIP**. Use [Figment](https://docs.rs/figment/latest/figment/) crate: https://github.com/torrust/torrust-tracker/pull/808

PR reviews: 

- https://github.com/torrust/torrust-tracker/pull/810

### Torrust Index

New Issues: 

- [ ] https://github.com/torrust/torrust-index/issues/567
- [ ] https://github.com/torrust/torrust-index/issues/562

### Torrust Index GUI

New Issues: 

- [ ] https://github.com/torrust/torrust-index-gui/issues/529
- [ ] https://github.com/torrust/torrust-index-gui/issues/527

Comment on issues:

- [ ] https://github.com/torrust/torrust-index-gui/issues/524
- [ ] https://github.com/torrust/torrust-index-gui/issues/522

PR reviews: 

- https://github.com/torrust/torrust-index-gui/pull/536
- https://github.com/torrust/torrust-index-gui/pull/535

### Torrust Demo

I've been testing adding new sample torrents to the demo with different BitTorrent clients. I had problems for seeding with Deluge because it uses non-standard fields in the `info` dict. When that happens, you need to download the new torrent from the Index and try to change it in the client. However, telling the client you already have the sources for a new torrent seems tricky. I wasn't able to do it. [I've opened an issue about this](https://github.com/torrust/torrust-index/issues/567).

The demo server is also having problems. [It's crashing every 2 hours](https://github.com/torrust/torrust-demo/issues/1).

Issues:

- [ ] https://github.com/torrust/torrust-demo/issues/1

### Torrust Website

@cgbosse and I met with a designer working on the new design for the Torrust website.

PR reviews: 

- https://github.com/torrust/torrust-website/pull/162
- https://github.com/torrust/torrust-website/pull/159

### Other topics

We have a new contributor, @ngthhu, who is opening a lot of PRs. Thank you @ngthhu!

**Comment by grmbyrn** [comment]
## Torrust Website
* Fix the Storybook as in [#104](https://github.com/torrust/torrust-website/issues/104)
* Update Svelte version
* Uninstall deprecated [eslint-plugin-svelte3](https://github.com/sveltejs/eslint-plugin-svelte3) and install recommended [eslint-plugin-svelte](https://github.com/sveltejs/eslint-plugin-svelte)
* Work on solving other failures in the build
* Found issues caused by moving to svelte@4.2.15, where my .eslintrc.cjs was still referencing some old "svelte3" configuration. Replace `.eslintrc.js` by copying latest config from a SvelteKit project generated with `npm create svelte@latest my-app`. From this:
```
module.exports = {
	root: true,
	extends: ['eslint:recommended', 'prettier', 'plugin:@typescript-eslint/recommended'],
	plugins: ['svelte3', '@typescript-eslint'],
	overrides: [
		{
			files: ['*.svelte'],
			processor: 'svelte3/svelte3'
		}
	],
	settings: {
		'svelte3/typescript': () => require('typescript') // pass the TypeScript package to the Svelte plugin
	},
	parserOptions: {
		sourceType: 'module',
		ecmaVersion: 2020
	},
	env: {
		browser: true,
		es2017: true,
		node: true
	}
};
```

to this:
```
module.exports = {
    root: true,
    extends: [
        'eslint:recommended',
        'plugin:@typescript-eslint/recommended',
        'plugin:svelte/recommended',
        'prettier',
    ],
    parser: '@typescript-eslint/parser',
    plugins: ['@typescript-eslint'],
    parserOptions: {
        sourceType: 'module',
        ecmaVersion: 2020,
        extraFileExtensions: ['.svelte'],
    },
    env: {
        browser: true,
        es2017: true,
        node: true,
    },
    overrides: [
        {
            files: ['*.svelte'],
            parser: 'svelte-eslint-parser',
            parserOptions: {
                parser: '@typescript-eslint/parser',
            },
        },
    ],
};
```

## Torrust
* Set up dev env. Meeting with Jose to discuss problems with setting it up

## Torrust-Index-GUI
* [Manually upload file not working](https://github.com/torrust/torrust-index-gui/issues/532) - updated Chrome version and it worked.
* Change the upload section so that user can click anywhere inside dashed border to upload file, not just on `Manually upload a file`

![image](https://github.com/nautilus-cyberneering/.github/assets/95353365/f84ac8f0-8568-4761-a575-7d8ef3df6343)

## Rust
* Continue learning about Rust
